modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/utils/JavaMetadataKeyIdModelParserUtils.java

```diff
@@ -221,6 +221,13 @@ private static Optional<MetadataKeyModelParser> getKeyIdResolverModelParser(Outp
       }
     }
 
+    if (keyIdResolverModelParser.isPresent()
+        && outputResolverModelParser == null
+        && attributesResolverModelParser == null
+        && inputResolverModelParsers.isEmpty()) {
+      LOGGER.warn("A Keys Resolver is being defined without defining an Output Resolver, Input Resolver nor Attributes Resolver");
```

Comment:
The improvements for the log message are missing, see the partial work PR in the issue.